mlir/include/Quantum/IR/QuantumOps.td

```diff
@@ -356,54 +384,26 @@ def GlobalPhaseOp : Quantum_Op<"gphase", [DifferentiableGate, AttrSizedOperandSe
         `(` $params `)` attr-dict ( `ctrls` `(` $in_ctrl_qubits^ `)` )?  ( `ctrlvals` `(` $in_ctrl_values^ `)` )? `:` (`ctrls` type($out_ctrl_qubits)^ )?
     }];
 
-    code extraBaseClassDeclaration = [{
+    let extraClassDeclaration = extraBaseClassDeclaration # [{
         mlir::ValueRange getAllParams() {
             return getODSOperands(getParamOperandIdx());
         }
 
-        std::vector<mlir::Value> getQubitOperands() {
-            std::vector<mlir::Value> values;
-            values.insert(values.end(), getInCtrlQubits().begin(),
-                          getInCtrlQubits().end());
-            return values;
-        }
-
-        std::vector<mlir::Value> getQubitResults() {
-            std::vector<mlir::Value> values;
-            values.insert(values.end(), getOutCtrlQubits().begin(),
-                          getOutCtrlQubits().end());
-            return values;
-        }
-
-        bool getAdjointFlag() {
```

Comment:
All the adjoint flag stuff is not necessary?

Reply:
Also explained this above, but this Op now inherits from the `Gate_op` class so we don't need to redefine all of these :)